@pmaier1 the syntax 'tls internal' in the Caddyfile does not work for me. 

```
caddy_1           | 2020/08/27 16:01:13 [ERROR] Making new certificate manager: registration error: acme: error: 400 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-acct :: urn:ietf:params:acme:error:invalidEmail :: Error creating new account :: "internal" is not a valid e-mail address, url:  (attempt 2/3)
caddy_1           | 2020/08/27 16:01:13 [INFO] acme: Registering account for internal

caddy_1           | 2020/08/27 16:01:15 [ERROR] Making new certificate manager: registration error: acme: error: 400 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-acct :: urn:ietf:params:acme:error:invalidEmail :: Error creating new account :: "internal" is not a valid e-mail address, url:  (attempt 3/3)
caddy_1           | 2020/08/27 16:01:15 registration error: acme: error: 400 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-acct :: urn:ietf:params:acme:error:invalidEmail :: Error creating new account :: "internal" is not a valid e-mail address, url: 
caddy_1           | exit status 1
compose_caddy_1 exited with code 1
```
Caddy v1: tls self_signed
Caddy v2: tls internal

Are you running Caddy v2???

--- a/compose/kopano/konnect/Caddyfile
+++ b/compose/kopano/konnect/Caddyfile
@@ -3,7 +3,10 @@
 	log stdout
 
 	# https://caddyserver.com/docs/caddyfile/directives/tls
-	tls internal
+	# abiosoft/caddy:latest caddy_version 1.0.3 interprets 'tls internal' as an email-address and fails.
+	# v1: 'tsl self_signed' generate and use an untrusted, self-signed certificate in memory that lasts 7 days
+	# caddy/latest CADDY_VERSION: 2.1.1 understands tls internal.
+	tls self_signed
 
 	# konnect oidc
 	proxy /.well-known/openid-configuration kopano_konnect:8777
@@ -30,7 +33,7 @@
 	errors stderr
 	log stdout
 
-	tls internal
+	tls self_signed
 
 	# owncloud 10
 	proxy / owncloud:8080 {

--- a/compose/kopano/konnect/Caddyfile.v2
+++ b/compose/kopano/konnect/Caddyfile.v2
@@ -1,0 +1,76 @@
+{$KOPANO_KONNECT_DOMAIN}:80 {
+	# errors stderr
+	log
+
+	# konnect oidc
+	reverse_proxy /.well-known/openid-configuration kopano_konnect:8777
+	reverse_proxy /konnect/v1/jwks.json kopano_konnect:8777
+	reverse_proxy /konnect/v1/token kopano_konnect:8777
+	reverse_proxy /konnect/v1/userinfo kopano_konnect:8777
+	reverse_proxy /konnect/v1/static kopano_konnect:8777
+	reverse_proxy /konnect/v1/session kopano_konnect:8777
+	reverse_proxy /konnect/v1/register kopano_konnect:8777
+
+	# konnect identifier login area
+	reverse_proxy /signin/ kopano_konnect:8777 {
+#		transparent
+	}
+
+	# owncloud 10
+	reverse_proxy /oc10 owncloud:8080 {
+#		without /oc10
+#		transparent
+	}
+}
+
+{$KOPANO_KONNECT_DOMAIN}:443 {
+	# errors stderr
+	log
+
+	# https://caddyserver.com/docs/caddyfile/directives/tls
+	# abiosoft/caddy:latest caddy_version 1.0.3 interprets 'tls internal' as an email-address and fails.
+	# caddy/latest CADDY_VERSION: 2.1.1 understands tls internal.
+	tls internal
+
+	# konnect oidc
+	reverse_proxy /.well-known/openid-configuration kopano_konnect:8777
+	reverse_proxy /konnect/v1/jwks.json kopano_konnect:8777
+	reverse_proxy /konnect/v1/token kopano_konnect:8777
+	reverse_proxy /konnect/v1/userinfo kopano_konnect:8777
+	reverse_proxy /konnect/v1/static kopano_konnect:8777
+	reverse_proxy /konnect/v1/session kopano_konnect:8777
+	reverse_proxy /konnect/v1/register kopano_konnect:8777
+
+	# konnect identifier login area
+	reverse_proxy /signin/ kopano_konnect:8777 {
+#		transparent
+	}
+
+	# owncloud 10
+	reverse_proxy /oc10 owncloud:8080 {
+#		without /oc10
+#		transparent
+	}
+}
+
+{$OWNCLOUD_DOMAIN}:80 {
+	# errors stderr
+	log
+
+	# owncloud 10
+	reverse_proxy / owncloud:8080 {
+#		transparent
+	}
+}
+
+{$OWNCLOUD_DOMAIN}:443 {
+	# errors stderr
+	log
+
+	tls internal
+
+	# owncloud 10
+	reverse_proxy / owncloud:8080 {
+#		transparent
+	}
+}

--- a/compose/kopano/konnect/README.md
+++ b/compose/kopano/konnect/README.md
@@ -13,6 +13,9 @@ start an ownCloud instance with a Kopano IDP:
 export KOPANO_KONNECT_DOMAIN konnect.docker-playground.local
 export OWNCLOUD_DOMAIN owncloud.docker-playground.local
 
+docker system prune -f
+docker volume prune -f
+
 docker-compose \
     -f owncloud-base.yml \
     -f owncloud-official.yml \

--- a/compose/kopano/konnect/README.md
+++ b/compose/kopano/konnect/README.md
@@ -34,7 +34,7 @@ docker-compose \
 Then manually sync LDAP users
 
 ```
-occ user:sync "OCA\User_LDAP\User_Proxy"
+occ user:sync 'OCA\User_LDAP\User_Proxy'
 ```
 
 And enable te openidconnect app

--- a/compose/kopano/konnect/README.md
+++ b/compose/kopano/konnect/README.md
@@ -10,9 +10,11 @@ start an ownCloud instance with a Kopano IDP:
 
 
 ```console
-export KOPANO_KONNECT_DOMAIN konnect.docker-playground.local
-export OWNCLOUD_DOMAIN owncloud.docker-playground.local
-
+export KOPANO_KONNECT_DOMAIN=konnect.docker-playground.local
+export OWNCLOUD_DOMAIN=owncloud.docker-playground.local
+echo >> /etc/hosts 127.0.0.1 $KOPANO_KONNECT_DOMAIN
+echo >> /etc/hosts 127.0.0.1 $OWNCLOUD_DOMAIN
+    
 docker system prune -f
 docker volume prune -f
 

--- a/compose/kopano/konnect/docker-compose.yml
+++ b/compose/kopano/konnect/docker-compose.yml
@@ -54,10 +54,14 @@ services:
       - /tmp
 
   caddy:
+    ## abiosoft/caddy:latest: is caddy_version: 1.0.3 -- cannot do: 'tls internal' is an invalid email, and 'tls off' causes ssl3_get_record:wrong version number
     image: "abiosoft/caddy:latest"
+    ## caddy:latest: is CADDY_VERSION: v2.1.1 -- 
+    # image: "caddy:latest"
     volumes:
       - ./.caddy:/root/.caddy # to sync mkcert certificates to Caddy
-      - ./kopano/konnect/Caddyfile:/etc/Caddyfile  # to mount custom Caddyfile
+      - ./kopano/konnect/Caddyfile:/etc/Caddyfile  # caddy 1.0.3: to mount custom Caddyfile
+      # - ./kopano/konnect/Caddyfile.v2:/etc/caddy/Caddyfile  # caddy 2.1.1: to mount custom Caddyfile
     ports:
       - 80:80
       - 443:443
@@ -65,6 +69,7 @@ services:
     environment:
       - ACME_AGREE=true
       - KOPANO_KONNECT_DOMAIN=${KOPANO_KONNECT_DOMAIN}
+      - OWNCLOUD_DOMAIN=${OWNCLOUD_DOMAIN}
     networks: 
       default:
         aliases: 

